src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -1875,6 +1876,47 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
             }
         }
 
+        public class ConstantSupport : JavascriptConversionExtension
+        {
+            public static readonly ConstantSupport Instance = new ConstantSupport();
+
+            private ConstantSupport()
+            {
+            }
+
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                if (!(context.Node is ConstantExpression ce))
+                    return;
+
+                var isDateTime = ce.Type == typeof(DateTime);
```

Comment:
Would be better to do using a case here, no?

Reply:
changed